### PR TITLE
Checkpoint 4 - Write a tweet on the UI

### PR DESCRIPTION
**Pull request by hritik5102**

### CODE REVIEW

File: checkpoints/fourth.patch [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ 
+   const getCharLimitIndicatorColor = () => {
+-    if (textArea.length > 200 && textArea.length < TEXT_CHAR_LIMIT) {
+-      return "text-yellow-600";
+-    }
+-
+-    if (textArea.length >= TEXT_CHAR_LIMIT) {
+-      return "text-red-600";
+-    }
++    // Task 2C : Make charachter limit indicator to red when user reaches the limit?
```
Comment (nrkin) [inline checkpoints/fourth.patch]: character

File: checkpoints/fourth.patch [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+-            minRows={1}
+-            maxRows={10}
+-            value={textArea}
+-            onChange={onTextAreaChange}
+-            placeholder="What's happening?"
+-            className="w-full rounded text-xl px-4 py-2 border-gray-600 resize-none outline-none"
+-          />
++          {/* 
++              Task 1A: Create auto-resizable text input fields.
++              Hint: You can also utilize an external library for this task.
```
Comment (nrkin) [inline checkpoints/fourth.patch]: Should we be allowing adding external libraries?

File: checkpoints/fourth.patch [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+-    }
++    // Task 2D: Restrict the end user from writing any more characters when they reach the maximum number of characters allowed.
+ 
+     setTextArea(input);
+   };
+@@ -31,20 +29,14 @@ export default function ComposeTweet({ handle }) {
+   }, []);
+ 
+   const getCharLimitIndicatorColor = () => {
+-    if (textArea.length > 200 && textArea.length < TEXT_CHAR_LIMIT) {
```
Comment (hs-saunved) [inline checkpoints/fourth.patch]: This part can stay in the checkpoint

File: checkpoints/fourth.patch [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+-      return "text-yellow-600";
+-    }
+-
+-    if (textArea.length >= TEXT_CHAR_LIMIT) {
```
Comment (hs-saunved) [inline checkpoints/fourth.patch:20]: This part can be a todo

File: checkpoints/fourth.patch [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+-            router.push(`/${user.handle}/status/${body.data.tweet.id}`);
++            // Task 3B - Redirect the user to the single tweet view
+           });
+         } else {
+-          // If create tweet fails
+           response.json().then((body) => {
+-            if (body?.message) {
+-              toast.error(body.message);
+-            } else {
+-              toast.error(uiText.somethingWentWrong);
```
Comment (hs-saunved) [inline checkpoints/fourth.patch:53]: Keep the if/else conditions visible. Just keep the toast.error portion a todo

File: checkpoints/fourth.patch [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+         <form onSubmit={onTweetSubmit} className="w-full">
+-          <ReactTextareaAutosize
+-            minRows={1}
+-            maxRows={10}
+-            value={textArea}
+-            onChange={onTextAreaChange}
+-            placeholder="What's happening?"
+-            className="w-full rounded text-xl px-4 py-2 border-gray-600 resize-none outline-none"
+-          />
++          {/*
```
Comment (hs-saunved) [inline checkpoints/fourth.patch:61]: We should just keep the onChange and value part a todo. The rest of this component should be pre-filled. I don't think we can expect students to figure this out on Day 1.

**Review by hs-saunved — CHANGES_REQUESTED** [review]

File: checkpoints/fourth.patch (hunk 2 above)
Comment (hs-saunved) [reply checkpoints/fourth.patch]: No, the library should be added from before. Check my comment above.

File: checkpoints/fourth.patch (hunk 3 above)
Comment (hritik5102) [reply checkpoints/fourth.patch]: Ack

File: checkpoints/fourth.patch (hunk 5 above)
Comment (hritik5102) [reply checkpoints/fourth.patch:53]: Ack

**Pull request closed without merging**